src/extension.ts

```diff
@@ -22,42 +79,163 @@ export function activate(context: vscode.ExtensionContext) {
+                ? vscode.l10n.t('passphraseInputPromptTitleWhenSecurelyPassphraseCacheEnabled')
+                : vscode.l10n.t('passphraseInputPromptTitle'),
             password: true,
+            placeHolder: keyStatusManager.currentKey.userId
+                ? vscode.l10n.t('keyDescriptionWithUsedId', keyStatusManager.currentKey.userId)
+                : undefined,
         });
         if (passphrase === undefined) { return; }
         try {
             await keyStatusManager.unlockCurrentKey(passphrase);
-            vscode.window.showInformationMessage('Key unlocked.');
+            if (keyStatusManager.enableSecurelyPassphraseCache) {
+                await secretStorage.set(keyStatusManager.currentKey.fingerprint, passphrase);
+                vscode.window.showInformationMessage(vscode.l10n.t('keyUnlockedWithCachedPassphrase'));
+            } else {
+                vscode.window.showInformationMessage(vscode.l10n.t('keyUnlocked'));
+            }
+            await keyStatusManager.syncStatus();
         } catch (err) {
             if (err instanceof Error) {
-                vscode.window.showErrorMessage(`Failed to unlock: ${err.message}`);
+                vscode.window.showErrorMessage(vscode.l10n.t('keyUnlockFailed', err.message));
             }
         }
-    });
-    context.subscriptions.push(command);
+    }));
     keyStatusItem.tooltip = 'Unlock this key';
     keyStatusItem.command = commandId;
 
-    if (vscode.workspace.workspaceFolders !== undefined) {
-        const folders = toFolders(vscode.workspace.workspaceFolders);
-        keyStatusManager.updateFolders(folders);
-        keyStatusManager.changeActivateFolder(folders[0]);
-    }
+    context.subscriptions.push(vscode.commands.registerCommand("gpgIndicator.deletePassphraseCache", async () => {
+        const items: vscode.QuickPickItem[] | false = await generateKeyList(secretStorage, keyStatusManager);
+        if (!items) {
+            vscode.window.showInformationMessage(vscode.l10n.t('noCachedPassphrase'));
+            return;
+        }
+        const targets = await vscode.window.showQuickPick(items, {
+            title: vscode.l10n.t("cachedPassphraseListForDeletion"),
+            canPickMany: true,
+            ignoreFocusOut: true,
+            matchOnDescription: true,
+            matchOnDetail: true,
+            placeHolder: vscode.l10n.t("cachedPassphraseListForDeletionPlaceHolder"),
+        });
+        if (!Array.isArray(targets) || targets.length === 0) {
+            return;
+        }
+        if ((await vscode.window.showInformationMessage<vscode.MessageItem>(
+            vscode.l10n.t("passphraseDeletionConfirm"),
+            { modal: true, detail: vscode.l10n.t('yourSelection', targets.map(({ label, detail }) => `* ${label}${detail ? `\n  ${detail}` : ""}`).join('\n')) },
+            { title: YES },
+            { title: NO, isCloseAffordance: true },
+        ))?.title !== YES) {
+            return;
+        }
```

Comment:
> For the third comment, I think the user might want to be able to delete the cache, for example when the user abandoned their old keys, maybe they want to delete the cached passphrase, even if it's stored in SecretStorage.

Agreed. What I means is that we keep `gpgIndicator.deletePassphraseCache` command but remove the double checking of `passphraseDeletionConfirm` dialog. :D

Same for `gpgIndicator.clearPassphraseCache` and `passphraseClearanceConfirm` below.

Reply:
I agreed for removing `passphraseDeletionConfirm`, but I think the `passphraseClearanceConfirm` should be kept for possible false click in the command list.

Reply:
OK :D